src/Model/MethodGo.cs

```diff
@@ -1,699 +1,752 @@
+            // we will continue embedding that in each method to have broader support.
+            APIVersion = apiVersionParam.SingleOrDefault();
+            if (APIVersion == default(string))
+            {
+                APIVersion = cmg.ApiVersion;
+            }
+
+            var parameter = Parameters.ToList().Find(p => p.ModelType.PrimaryType(KnownPrimaryType.Stream)
+                                                && !(p.Location == ParameterLocation.Body || p.Location == ParameterLocation.FormData));
+
+            if (parameter != null)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                    Resources.IllegalStreamingParameter, parameter.Name));
+            }
+            if (string.IsNullOrEmpty(Description))
+            {
+                Description = string.Format("sends the {0} request.", Name.ToString().ToPhrase());
+            }
+
+            // Registering Azure resource providers should only happen with Azure resource manager REST APIs
+            // This depends on go-autorest here:
+            // https://github.com/Azure/go-autorest/blob/c0eb859387e57a164bf64171da307e2ef8168b58/autorest/azure/rp.go#L30
+            // As registering needs the Azure subscription ID, we take it from the operation path, on the
+            // assumption that ARM APIs should include the subscription ID right after `subscriptions`
+            RegisterRP = cmg.APIType.EqualsIgnoreCase("arm") && Url.Split("/").Any(p => p.EqualsIgnoreCase("subscriptions"));
+
+            // Fixing the returnType in modeler
+            // find all the non-error responses with non-empty body
+            // we have to ignore those non-error responses with empty bodies, because there are quite plenty of swaggers with one response with body and the other without body
+            Logger.Instance.Log(Category.Debug, $"All responses of {SerializedName}: {string.Join(", ", Responses.Select(kv => $"{(int)kv.Key}: {kv.Value?.Body?.Name}"))}");
+            var nonErrorNonEmptyStatusCodes = NonErrorNonEmptyStatusCodeDict;
+            Logger.Instance.Log(Category.Debug, $"Method {SerializedName} has the following non-error & non-empty responses (total {nonErrorNonEmptyStatusCodes.Count()}): {string.Join(", ", nonErrorNonEmptyStatusCodes.Select(resp => resp.Value?.Body?.Name))}");
+            // categorize the models by its name
+            var nonErrorNonEmptyResponses = NonErrorNonEmptyResponses.ToList();
+            Logger.Instance.Log(Category.Debug, $"Non-error & non-empty models: {string.Join(", ", nonErrorNonEmptyResponses.Select(model => model.Body.Name))}");
+            // fix the original return type from the modeler
+            ReturnType = FixedReturnType();
+            Logger.Instance.Log(Category.Debug, $"Return Type of {SerializedName}: {ReturnType?.Body?.Name}");
+        }
+
+        public Response FixedReturnType()
```

Comment:
Can't we move this into `ReturnValue()`?  Now we have two ways of getting the return type, how do you know the right one is being invoked?

Reply:
Hmm, reasonable... let me refactor...

Reply:
Hi @jhendrixMSFT it turns out that I have to keep this function. instead, I removed the `ReturnValue` function because `ReturnType` could never be null and this makes it no difference from `ReturnType` (inherited from `Method`)
We also cannot override the property `ReturnType` using this function - otherwise this makes the setting of the `ReturnType` meaningless.
In the transform process, we need to keep the order of the operation unchanged, otherwise we might get unexpected result.
Previously, the transformation of the methods happens after the transformation of models. Therefore in the model transformation, it is better that we only use the transformed value for this transformation, and do not introduce the transformation prematurely.
Based on the above reasons, I believe we need to keep this.

Reply:
To clarify, we should never use this function, unless we are in the transformation process. In this way we can know which one is the right one to invoke - the right one is always the `ReturnType` inherited from `Method`

Reply:
and FYI I am regenerating every package in the track 1 sdk now to see if there is any bad impact

Reply:
Ok, if we need to keep it and it's only used during transformation, can you please add a doc comment to this method explaining that?

I'll be watching for an update on your tests.

Reply:
I just added a comment on this function